### ISSUE TITLE
False positive with pip install versioon

### ISSUE THREAD

**Issue by xtibau** [hadolint/hadolint#915]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior
Trying to avoid the pip install . warning (see here)[https://github.com/pypa/pip/issues/8559#issuecomment-1365250852] I do get a wrong warning. 

### Actual behavior
I do get  `DL3013 warning: Pin versions in pip. Instead of `pip install <package>` use `pip install <package>==<version>` or `pip install --requirement <requirements file>`

### Steps to reproduce the behavior

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
run --rm -i hadolint/hadolint < package.Dockerfile
```
output:
``` bash
-:4 DL3008 warning: Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`
-:18 DL3013 warning: Pin versions in pip. Instead of `pip install <package>` use `pip install <package>==<version>` or `pip install --requirement <requirements file>`
```

### Notes
It does only happen because I use only one RUN. Also note that I also get another warning when installing the gcc but I'm unable to fix that :\

**Comment by kthy** [comment]
Please post your Dockerfile.
